src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[3];
+  const oldDateFormat = {};
+  const separatedDate = date.split(separator);
+
+  fromFormat.slice(0, 3).forEach((key, index) => {
+    oldDateFormat[key] = separatedDate[index];
+  });
+
+  if (fromFormat.includes('YY')) {
+    oldDateFormat['YYYY'] =
+      parseInt(oldDateFormat['YY'], 10) < 30
+        ? '20' + oldDateFormat['YY']
+        : '19' + oldDateFormat['YY'];
+  } else if (fromFormat.includes('YYYY')) {
+    oldDateFormat['YY'] = oldDateFormat['YYYY'].slice(-2);
+  }
+
+  const newSeparator = toFormat[3];
```

Comment:
Similar to the `fromFormat`, the separator is assumed to be at index 3 of the `toFormat` array. Ensure this assumption is valid for all expected output formats.